Lib/logging/__init__.py

```diff
@@ -237,19 +234,17 @@ def _acquireLock():
 
     This should be released with _releaseLock().
     """
-    if _lock:
-        try:
-            _lock.acquire()
-        except BaseException:
-            _lock.release()
-            raise
+    try:
+        _lock.acquire()
+    except BaseException:
+        _lock.release()
```

Comment:
I don't get this except: release() block. Does someone know the rationale?

Reply:
Yes, it's because of gh-106238.

Reply:
This code is very surprising and would deserve a comment on its own. The relationship with register_at_fork() is not obvious at all.

@dcollison: Do you want to take the opportunity to add a comment, at least mention gh-106238.

Reply:
With the whole PR, the last caller of _acquireLock() is now register_at_fork() below. Would you mind to rewrite the docstring and rename the function?

You can just rename the function to _prepareFork(). And _releaseLock() can be renamed to _afterFork().

Reply:
Sure :) I've made the requested changes.  Is it safe to rename _acquireLock and _releaseLock?

Reply:
It's a private method it should not be used outside the stdlib.

Reply:
> how about _prepare_fork & _after_fork?

The rest of the module/package uses camelCase, so it's probably best to keep the naming consistent.